### DISCUSSION TITLE
WARNING: SU2 was not compiled for an AVX-capable architecture.

### DISCUSSION THREAD

**Discussion by aditya12398** [su2code/SU2#1853] [Q&A] [answered]
I apologize in advanced if this issue seems to be trivial or stupid but I genuinely need help.

**Describe the issue**

Trying to run the Onera-M6 Newton-Krylov method when the solver starts with aforementioned warning.

I have a `AMD Ryzen 5 3600` CPU which supports avx and avx2 architectures so this warning seems a little off somehow.

I did check out this [CFD-Online Post](https://www.cfd-online.com/Forums/su2-installation/244862-su2-not-installed-avx-architecture.html). However, [this page on Gentoo wiki](https://wiki.gentoo.org/wiki/GCC_optimization) suggests against using `-march=native` as a CXXFLAG for compiling.

`gcc -c -Q -march=native --help=target` shows `-mavx` and `-mavx2` enabled but when I do `gcc -c -Q -march=ryzen --help=target` those two options are disabled. I can use `x86-64-v3` instead which, [according to this answer/script](https://unix.stackexchange.com/questions/631217/how-do-i-check-if-my-cpu-supports-x86-64-v2), is supported by my CPU but at this point I am just extremely confused as to what to do and how to avoid this warning.

Shall I recompile SU2 with this `-march` flag? If yes, which flag should I use out of these `ryzen`, `x86-64`, `native`, or `x86-64-v3`? Or is it something else altogether

**Desktop information:**
 - OS: Arch Linux
 - CPU: AMD Ryzen 5 3600 (12)
 - C++ compiler and version: g++ (GCC) 12.2.0
 - MPI implementation and version: MPICH 4.0.3
 - SU2 Version: 7.4.0

Kindly let me know if you need any other details. Thanks!

**Comment by pcarruscag** [comment, marked as answer]
This is what I use to run locally
`export CXXFLAGS="-march=native -funroll-loops -ffast-math -fno-finite-math-only"` with `--optimization=2`

**Reply by aditya12398** [reply]
Oh, thanks a lot!

Will this option help get rid of the warning?

Also, what is the reason for using `--optimization=2` instead of 3?
